Add name to job:slo_batch_throughput_target:max

dyson opened the pull request:
Currently each batch processing SLO creates a recording rule that looks like this:
```
record: job:slo_batch_throughput_target:max
expr: job:slo_batch_volume:max
  / 600
labels:
  name: GenerateEURMerchantPayoutsMeetsDeadline
```

This expression is relying on another recording rule that all batch
processing rule SLOs create, `job:slo_batch_volume:max`.

`job:slo_batch_volume:max` contains all the results for all rules, eg:

```
{currency="EUR",name="MarkEURPaymentsAsPaidMeetsDeadline"}
{currency="GBP",name="GenerateGBPMerchantPayoutsMeetsDeadline"}
```

As seen above in `job:slo_batch_throughput_target:max` we apply the SLO
label name and as all results are returned we will overwrite all name labels.

This rightly leads to the error:
`vector contains metrics with the same labelset after applying rule
labels`

We need to ensure that we filter the results of job:slo_batch_volume:max
by the name of the SLO to prevent this and record correctly.

eg:
```
record: job:slo_batch_throughput_target:max
expr: job:slo_batch_volume:max{name="GenerateEURMerchantPayoutsMeetsDeadline"}
  / 600
labels:
  name: GenerateEURMerchantPayoutsMeetsDeadline
```

dyson committed:
Add name to job:slo_batch_throughput_target:max

Currently each batch processibg SLO creates a recording rule that looks like this:
```
record: job:slo_batch_throughput_target:max
expr: job:slo_batch_volume:max
  / 600
labels:
  name: GenerateEURMerchantPayoutsMeetsDeadline
```

This expression is relying on another recoring rule that all batch
processing rule SLOs create, `job:slo_batch_volume:max`.

`job:slo_batch_volume:max` contains all the results for all rules, eg:

```
{currency="EUR",name="MarkEURPaymentsAsPaidMeetsDeadline"}
{currency="GBP",name="GenerateGBPMerchantPayoutsMeetsDeadline"}
```

As seen above in `job:slo_batch_throughput_target:max` we apply the SLO
label name and as all results are returned we will overwrite all name labels.

This rightly leads to the error:
`vector contains metrics with the same labelset after applying rule
labels`

We need to ensure that we filter the results of job:slo_batch_volume:max
by the name of the SLO to prevent this and record correctly.

eg:
```
record: job:slo_batch_throughput_target:max
expr: job:slo_batch_volume:max{name="GenerateEURMerchantPayoutsMeetsDeadline"}
  / 600
labels:
  name: GenerateEURMerchantPayoutsMeetsDeadline
```

benwh reviewed (commented):
The explanation and code change definitely makes sense!

Would be good to see the tests passing (some circle-fixing to do?), and maybe also the changes that it would produce in the anu SLO builder definitions?

dyson commented:
Great! Will do all that now.

dyson committed:
Bump go and prometheus in circleci

dyson commented:
Should be good now for another review. 

- Fixed circle
- Updated the example so you can see the output
- Bumped prometheus and go while I was there

benwh reviewed (approved):
Nice! LGTM
